diff --git a/src/resources/lib/modules/system.py b/src/resources/lib/modules/system.py
index 2b53cbc..f785367 100644
--- a/src/resources/lib/modules/system.py
+++ b/src/resources/lib/modules/system.py
@@ -199,14 +199,6 @@ class system:
                             'InfoText': 724,
                             'order': 1,
                             },
-                        'oe_reset': {
-                            'name': 32325,
-                            'value': '0',
-                            'action': 'reset_oe',
-                            'type': 'button',
-                            'InfoText': 725,
-                            'order': 2,
-                            },
                         },
                     },
                 'switch': {
@@ -634,23 +626,6 @@ class system:
             self.oe.set_busy(0)
             self.oe.dbg_log('system::reset_xbmc', 'ERROR: (' + repr(e) + ')', 4)
 
-    def reset_oe(self, listItem=None):
-        try:
-            self.oe.dbg_log('system::reset_oe', 'enter_function', 0)
-            if self.ask_sure_reset('LibreELEC') == 1:
-                self.oe.set_busy(1)
-                reset_file = open(self.LIBREELEC_RESET_FILE, 'w')
-                reset_file.write('reset')
-                reset_file.close()
-                self.oe.winOeMain.close()
-                time.sleep(1)
-                xbmc.executebuiltin('Reboot')
-                self.oe.set_busy(0)
-            self.oe.dbg_log('system::reset_oe', 'exit_function', 0)
-        except Exception, e:
-            self.oe.set_busy(0)
-            self.oe.dbg_log('system::reset_oe', 'ERROR: (' + repr(e) + ')', 4)
-
     def switch_system(self, listItem=None):
         try:
             self.oe.dbg_log('system::switch_system', 'enter_function', 0)
-- 
1.7.10.4

