--- linux-83ba060.orig/arch/arm/boot/dts/amlogic/meson8_t8_2G.dtd	2016-06-11 15:59:55.210940179 +0200
+++ linux-83ba060/arch/arm/boot/dts/amlogic/meson8_t8_2G.dtd	2016-06-11 16:04:36.000000000 +0200
@@ -133,11 +133,11 @@
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x01800000  0x00100000>;
+		reserve-memory = <0x04800000  0x00100000>;
 		reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
 		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 
@@ -333,6 +333,19 @@
 	};
 
 /// ***************************************************************************************
+///    -       Vfd
+//$$ DEVICE="m1-vfd"
+//$$ L2 PROP_STR = "status"
+    m1-vfd{
+               compatible = "amlogic,m1-vfd";
+               dev_name = "m1-vfd.0";
+               status = "okay";
+               vfd_stb_pin = "GPIOH_3";
+               vfd_clk_pin =  "GPIOH_6";
+               vfd_data_pin = "GPIOH_5";
+       };
+
+/// ***************************************************************************************
 ///	-	UART
 //$$ MODULE="UART"
 //$$ DEVICE="uart_ao"
@@ -491,7 +504,7 @@
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+            //gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -803,7 +816,7 @@
 			port-config = <0>; /** 0: default */
 			port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
 			port-id-mode = <0>; /** 0: hardware, 1: sw_host, 2: sw_slave*/
-			gpio-vbus-power = "GPIOZ_1";
+			gpio-vbus-power = "GPIOY_14";
 			gpio-work-mask	= <1>; /**0: work on pulldown,1:work on pullup*/
 			status = "okay";
 		};
